pynitrokey/nethsm/__init__.py

```diff
@@ -176,23 +176,49 @@ def __init__(
 
 
 def _handle_api_exception(e, messages={}, roles=[], state=None):
+    # give priority to custom messages
     if e.status in messages:
         message = messages[e.status]
+        raise NetHSMError(message)
+
+    if e.status == 401 and roles:
+        message = "Unauthorized -- invalid username or password"
     elif e.status == 403 and roles:
         roles = [role.value for role in roles]
         message = "Access denied -- this operation requires the role " + " or ".join(
             roles
         )
-    elif e.status == 401 and roles:
-        message = "Unauthorized -- invalid username or password"
+    elif e.status == 405:
+        # 405 "Method Not Allowed" mostly happens when the UserID or KeyID contains a character
+        # - that ends the path of the URL like a question mark '?' :
+        #   /api/v1/keys/?/cert will hit the keys listing endpoint instead of the key/{KeyID}/cert endpoint
+        # - that doesn't cound as a phath parameter like a slash '/' :
+        #   /api/v1/keys///cert will be interpreted as /api/v1/keys/cert with cert as the KeyID
```

Comment:
Is that something that we should validate before sending the request?

Reply:
That would be the best solution yes

Reply:
Please create a ticket for it (or a PR ;)).